_studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp

```diff
@@ -3417,7 +3417,12 @@ mfxStatus MfxHwH264Encode::CheckVideoParamQueryLike(
 
     if (extOpt2->LookAheadDS > MFX_LOOKAHEAD_DS_OFF)
     {
-        par.calcParam.widthLa  = mfx::align2_value((par.mfx.FrameInfo.Width  / LaDSenumToFactor(extOpt2->LookAheadDS)), 16);
+        if (extOpt2->LookAheadDS > MFX_LOOKAHEAD_DS_4x)
+        {
+            extOpt2->LookAheadDS = MFX_LOOKAHEAD_DS_2x;
```

Comment:
no. If you adjust it to some valid value - it is ```changed```. So if you can make some meaningful correction it should be something:
0)
```
if (extOpt2->LookAheadDS > MFX_LOOKAHEAD_DS_4x)
{
    extOpt2->LookAheadDS = MFX_LOOKAHEAD_DS_2x; // maybe MFX_LOOKAHEAD_DS_4x better? User wanted maximal allowed DS
    changed = true;
}
```

If you can't adjust - you have 2 options:
1) return immediately with error which means that you can't proceed encoding anyway
2) set value to 0 (here it would be ```MFX_LOOKAHEAD_DS_UNKNOWN```) and make ```unsupported``` true

So you should choose either 0) either 2). I think 0) is better

Reply:
I prefer 0) too.
But MFX_LOOKAHEAD_DS_4x isn't better, because it is replaced with MFX_LOOKAHEAD_DS_2x
https://github.com/Intel-Media-SDK/MediaSDK/blob/f5875364dd55a2efdd69540fd059722361f528bd/_studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp#L3412
`if (extOpt2->LookAheadDS == MFX_LOOKAHEAD_DS_4x)
    {
        extOpt2->LookAheadDS = MFX_LOOKAHEAD_DS_2x;
        changed = true;
    }`

Reply:
Then you need to reuse this logic in following way:
```
    if (extOpt2->LookAheadDS > MFX_LOOKAHEAD_DS_4x)
    {
        extOpt2->LookAheadDS = MFX_LOOKAHEAD_DS_4x;
        changed = true;
    }

    if (extOpt2->LookAheadDS == MFX_LOOKAHEAD_DS_4x)
    {
        extOpt2->LookAheadDS = MFX_LOOKAHEAD_DS_2x;
        changed = true;
    }

    if (extOpt2->LookAheadDS > MFX_LOOKAHEAD_DS_OFF)
    {
        par.calcParam.widthLa  = mfx::align2_value((par.mfx.FrameInfo.Width  / LaDSenumToFactor(extOpt2->LookAheadDS)), 16);
        par.calcParam.heightLa = mfx::align2_value((par.mfx.FrameInfo.Height / LaDSenumToFactor(extOpt2->LookAheadDS)), 16);
    } else
    {
        par.calcParam.widthLa  = par.mfx.FrameInfo.Width;
        par.calcParam.heightLa = par.mfx.FrameInfo.Height;
    }
```

the benefit is that if in future 4x will be supported it would require only one change in code